app/controllers/sessions_controller.rb

```diff
@@ -3,18 +3,37 @@ def new; end
 
   def create
     user = User.find_by email: params.dig(:session, :email)&.downcase
-    if user&.authenticate params.dig(:session, :password)
-      reset_session
-      log_in user
+    authenticate_user(user)
+  end
+
+  def destroy
+    log_out if logged_in?
+    redirect_to root_url
+  end
+
+  private
+
+  def authenticate_user user
+    if user&.authenticate(params[:session][:password])
+      reset_and_remember(user)
       redirect_to user
     else
-      flash.now[:danger] = t "invalid_email_password_combination"
-      render :new, status: :forbidden
+      handle_invalid_login
     end
   end
 
-  def destroy
-    log_out
-    redirect_to root_url
+  def handle_invalid_login
+    flash.now[:danger] = t("invalid_email_password_combination")
+    render :new, status: :forbidden
+  end
+
+  def reset_and_remember user
+    reset_session
+    remember_or_forget user
+    log_in(user)
```

Comment:
```suggestion
    log_in user
```
hạn chế dùng ngoặc trong ruby nha